File: modules/swagger-core/src/test/scala/1_3/converter/HiddenFieldTest.scala
package converter

import io.swagger.annotations.ApiModelProperty
import io.swagger.converter.ModelConverters
import io.swagger.models.properties.{LongProperty, StringProperty}
import org.junit.runner.RunWith
import org.scalatest.{FlatSpec, Matchers}
import org.scalatest.junit.JUnitRunner

import scala.beans.BeanProperty

@RunWith(classOf[JUnitRunner])
class HiddenFieldTest extends FlatSpec with Matchers {
  it should "ignore a hidden field" in {
    val models = ModelConverters.getInstance().read(classOf[ModelWithHiddenFields])

    val model = models.get("ModelWithHiddenFields")
    model should not be (null)
    model.getProperties.size should be(2)

    val idValue = model.getProperties.get("id")
    idValue.isInstanceOf[LongProperty] should be(true)
    idValue.getRequired should be(true)

    val nameValue = model.getProperties.get("name")
    nameValue.isInstanceOf[StringProperty] should be(true)
  }
}

class ModelWithHiddenFields {
  @BeanProperty
  @ApiModelProperty(required = true)
  var id: Long = _

  @BeanProperty
  @ApiModelProperty(required = true, hidden = false)
  var name: String = _

  @BeanProperty
  @ApiModelProperty(hidden = true)
  var password: String = _
}

File: modules/swagger-jersey2-jaxrs/src/test/java/io/swagger/models/HiddenFieldTest.java
package io.swagger.models;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModelProperty;
import io.swagger.converter.ModelConverters;
import io.swagger.models.Model;
import io.swagger.models.properties.LongProperty;
import io.swagger.models.properties.Property;
import io.swagger.models.properties.StringProperty;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import org.testng.annotations.Test;

import java.util.Map;

import static org.testng.Assert.*;

public class HiddenFieldTest {

    private final static int NUMBER_OF_PROPERTIES = 2;

    @Test(description = "it shouldn't ignore a hidden field")
    public void testHiddenField() {
        final Map<String, Model> models = ModelConverters.getInstance().read(ModelWithHiddenFields.class);

        final Model model = models.get("ModelWithHiddenFields");
        assertNotNull(model);
        assertEquals(model.getProperties().size(), NUMBER_OF_PROPERTIES);

        final Property idValue = model.getProperties().get("id");
        assertTrue(idValue instanceof LongProperty);
        assertTrue(idValue.getRequired());

        final Property nameValue = model.getProperties().get("name");
        assertTrue(nameValue instanceof StringProperty);

        final Property passwordValue = model.getProperties().get("password");
        assertNull(passwordValue);

        final Property emailValue = model.getProperties().get("email");
        assertNull(emailValue);
    }

    @Getter
    @Setter
    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    class ModelWithHiddenFields {
        @ApiModelProperty(required = true)
        public Long id = null;

        @ApiModelProperty(required = true, hidden = false)
        public String name = null;

        @ApiModelProperty(hidden = true)
        public String password = null;

        @ApiModelProperty(value = "Required Email", required = true, hidden = true)
        public String email = null;

        @JsonCreator
        private ModelWithHiddenFields(@JsonProperty("id") Long id,
                                      @JsonProperty("name") String name,
                                      @JsonProperty("password") String password,
                                      @JsonProperty("email") String email) {
            this.id = id;
            this.name = name;
            this.password = password;
            this.email = email;
        }
    }
}
